desc/compute/utils.py

```diff
@@ -91,6 +95,35 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
         data=data,
         **kwargs,
     )
+
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    parameterization = _parse_parameterization(parameterization)
+    for name in data.keys():
```

Comment:
can we swap the order of the loop and the `if basis == xyz` part? That way we won't have to check every qty if we're not doing any conversion